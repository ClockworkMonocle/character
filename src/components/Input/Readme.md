A simple style wrapper around an HTML input. Any props that can be applied to a normal `<input />` tag, can be applied here, except for `type`;

Basic example:

```js
<Input placeholder="type something..." />
```

Disabled:

```js
<Input disabled value="I'm disabled"/>
```

Read-Only:

```js
<Input readOnly value="Can select but cannot change"/>
```

Email input:

```js
<Input type="email" placeholder="type your email" />
```

Phone number input:

```js
<Input type="phone" placeholder="type your phone number" />
```

 Number input:

```js
<Input type="number" />
```
