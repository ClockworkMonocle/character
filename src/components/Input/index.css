
.input {
  display: block;
  width: 100%;
  appearance: none;
  border: 0;
  border-bottom: 1px solid var(--black);
  padding: 8px;
  outline: none;
  font-size: 1em;
  transition: border-bottom-color 300ms ease;
  background-color: transparent;
  border-radius: 0;
}

.input:focus,
.input:active {
  border-bottom-color: var(--attack-color);
}

.input:read-only {
  cursor: text;
}

.input:disabled {
  cursor: not-allowed;
}

.input_label span {
  font-size: 0.70em;
  font-weight: bold;
  text-transform: uppercase;
  color: var(--attack-color);
}
