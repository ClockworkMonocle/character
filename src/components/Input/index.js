import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './index.css';

export default function Input({ className, type, label, id, ...props }) {
  return <label htmlFor={id} className="input_label">
    {label && <span>{label}</span>}
    <input
      id={id}
      className={cn('input', className)}
      type={type}
      {...props}
    />
  </label>;
}

Input.propTypes = {
  type: PropTypes.oneOf(['email', 'text', 'phone', 'number']).isRequired,
  label: PropTypes.string,
};

Input.defaultProps = {
  type: 'text',
};
