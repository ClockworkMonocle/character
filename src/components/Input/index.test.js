import React from 'react';
import { shallow } from 'enzyme';
import Input from './index';

test('<Input /> renders text by default', () => {
  expect(shallow(<Input />)).toMatchSnapshot();
});

test('<Input /> renders with label', () => {
  expect(shallow(<Input label="Dis cool?" />)).toMatchSnapshot();
})
