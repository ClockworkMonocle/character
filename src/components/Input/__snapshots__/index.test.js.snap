// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders text by default 1`] = `
<label
  className="input_label"
>
  <input
    className="input"
    type="text"
  />
</label>
`;

exports[`<Input /> renders with label 1`] = `
<label
  className="input_label"
>
  <span>
    Dis cool?
  </span>
  <input
    className="input"
    type="text"
  />
</label>
`;
